### ISSUE TITLE
how to run 'demo.py'

### ISSUE THREAD

**Issue by CGC1031** [microsoft/Recursive-Cascaded-Networks#19]
Thanks for your contributions in this field.

I have successfully run your code of 'train.py' and 'eval.py' .But the results of them don't have any pictures. I really want to get the images and flow field after registration. Can I get them from 'demo.py'？And how can I get **-f FIXED_IMAGE -m MOVING_IMAGE** to run 'demo.py'?

Thanks a lot.Looking forward to your reply.

**Comment by zsyzzsoft** [comment]
demo.py requires two raw images. Otherwise, this answer #14 may be helpful to you.

**Comment by hjl2009** [comment]
As a newbie, I still have some questions. Where to download the raw images of liver and brain of DICOM series? Should these images fits the 128*128*128 size as well ? Or perhaps you can share your raw images with us

**Comment by zsyzzsoft** [comment]
If you only want to visualize the results, I recommend making some changes to eval.py, e.g., adding "real_flow" to the keys (see demo.py line 95-107).

**Comment by CGC1031** [comment]
Hellow,how to run my training code  on two gpus at the same time ?
My code can successfully run on GPU 0,but it is quiet slow. So I tried to use two gpus 0 and 1.
`parser.add_argument('-g', '--gpu', type=str, default='0,1',help='Specifies gpu device(s)')`
But something went wrong.
![QQ图片20200529183656](https://user-images.githubusercontent.com/53649087/83252297-9356b900-a1dd-11ea-9f9e-0a60f7f4f136.png)
![QQ图片20200529184003](https://user-images.githubusercontent.com/53649087/83252348-a7021f80-a1dd-11ea-8410-e340d6f4976b.png)
![QQ图片20200529184039](https://user-images.githubusercontent.com/53649087/83252353-a9647980-a1dd-11ea-899a-dafa5b0ecab8.png)
![QQ图片20200529184105](https://user-images.githubusercontent.com/53649087/83252374-b2554b00-a1dd-11ea-8278-161ae8bce2f9.png)
According to the prompt, the error is found in the **class MultiGPUs in line 192**.
Therefore, I suspect that there is something wrong with the format when I specify GPU. Do you have any good Suggestions to specify two GPUs？Looking forward to hearing from you.

**Comment by zsyzzsoft** [comment]
Hi, have you resolved this issue? I can't replicate this error but it seems like no gpu is specified.

**Comment by CGC1031** [comment]
Thank you for your reply. I have solved the problem. There is something wrong with my code.
 



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "zsyzzsoft"<notifications@github.com&gt;; 
发送时间: 2020年6月9日(星期二) 凌晨4:34
收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 
抄送: "蔡港成"<244040023@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run &#39;demo.py&#39; (#19)





 
Hi, have you resolved this issue? I can't replicate this error but it seems like no gpu is specified.

**Comment by w19787** [comment]
> Thank you for your reply. I have solved the problem. There is something wrong with my code.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zsyzzsoft"<notifications@github.com&gt;; 发送时间: 2020年6月9日(星期二) 凌晨4:34 收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 抄送: "蔡港成"<244040023@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run &#39;demo.py&#39; (#19) Hi, have you resolved this issue? I can't replicate this error but it seems like no gpu is specified. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

@CGC1031 how do you fix the problem? i have exact same problem when running on 4 gpus, but work fine on 1 gpu.

**Comment by CGC1031** [comment]
I have removed the ’Batch Normalization‘ code that I write and it has resolved the problem.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "w19787"<notifications@github.com&gt;; 
发送时间: 2020年6月12日(星期五) 中午12:32
收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 
抄送: "蔡港成"<244040023@qq.com&gt;; "Mention"<mention@noreply.github.com&gt;; 
主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run &#39;demo.py&#39; (#19)





  
Thank you for your reply. I have solved the problem. There is something wrong with my code.
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zsyzzsoft"<notifications@github.com&gt;; 发送时间: 2020年6月9日(星期二) 凌晨4:34 收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 抄送: "蔡港成"<244040023@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run 'demo.py' (#19) Hi, have you resolved this issue? I can't replicate this error but it seems like no gpu is specified. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
  
@CGC1031 how do you fix the problem? i have exact same problem when running on 4 gpus, but work fine on 1 gpu.

**Comment by w19787** [comment]
> I have removed the ’Batch Normalization‘ code that I write and it has resolved the problem.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "w19787"<notifications@github.com&gt;; 发送时间: 2020年6月12日(星期五) 中午12:32 收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 抄送: "蔡港成"<244040023@qq.com&gt;; "Mention"<mention@noreply.github.com&gt;; 主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run &#39;demo.py&#39; (#19) Thank you for your reply. I have solved the problem. There is something wrong with my code. … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zsyzzsoft"<notifications@github.com&gt;; 发送时间: 2020年6月9日(星期二) 凌晨4:34 收件人: "microsoft/Recursive-Cascaded-Networks"<Recursive-Cascaded-Networks@noreply.github.com&gt;; 抄送: "蔡港成"<244040023@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 主题: Re: [microsoft/Recursive-Cascaded-Networks] how to run 'demo.py' (#19) Hi, have you resolved this issue? I can't replicate this error but it seems like no gpu is specified. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe. @CGC1031 how do you fix the problem? i have exact same problem when running on 4 gpus, but work fine on 1 gpu. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

thanks. it should be different case as mine. i migrate the project to tensorflow 2.0, it might be mis-usage of multigpu support of 2.0. 
by the way, do you train the project from beginning? is the performance result same as paper? my result of mDist on liver DS is about 13.0, it is lower than expectation.
